crates/trie/sparse/src/trie.rs

```diff
@@ -666,21 +666,23 @@ impl RevealedSparseTrie {
                     }
 
                     branch_child_buf.clear();
-                    for bit in CHILD_INDEX_RANGE {
+                    for bit in CHILD_INDEX_RANGE.rev() {
                         if state_mask.is_bit_set(bit) {
                             let mut child = path.clone();
                             child.push_unchecked(bit);
                             branch_child_buf.push(child);
                         }
                     }
 
-                    branch_value_stack_buf.clear();
-                    for child_path in &branch_child_buf {
+                    branch_value_stack_buf.resize(branch_child_buf.len(), Default::default());
+                    let mut added_children = false;
+                    for (i, child_path) in branch_child_buf.iter().enumerate() {
                         if rlp_node_stack.last().map_or(false, |e| &e.0 == child_path) {
                             let (_, child) = rlp_node_stack.pop().unwrap();
-                            branch_value_stack_buf.push(child);
+                            branch_value_stack_buf[branch_child_buf.len() - i - 1] = child;
+                            added_children = true;
                         } else {
-                            debug_assert!(branch_value_stack_buf.is_empty());
+                            debug_assert!(!added_children);
                             path_stack.push(path);
                             path_stack.extend(branch_child_buf.drain(..));
                             continue 'main
```

Comment:
why do we need this reverse?

Reply:
if children are pushed from `0` to `f`, then the first one to pop will be `f` and we will do lookups in a non-sorted order https://github.com/paradigmxyz/reth/blob/70e88b90702a88ebc4c598d2e802dc5845ab32af/crates/trie/sparse/src/trie.rs#L633-L637

Reply:
nvm

Reply:
didn't see ur reply 😅 , ye, makes sense

Reply:
added a comment